pkg/net/net_linux.go

```diff
@@ -220,3 +223,115 @@ func netDevicePCIAddress(netDevDir, netDevName string) *string {
 	pciAddr := filepath.Base(devPath)
 	return &pciAddr
 }
+
+func (nic *NIC) setNicAttrEthtool(ctx *context.Context, dev string) error {
+	path, _ := exec.LookPath("ethtool")
+	cmd := exec.Command(path, dev)
+	var out bytes.Buffer
+	cmd.Stdout = &out
+	err := cmd.Run()
+	if err != nil {
+		msg := fmt.Sprintf("could not grab NIC link info for %s: %s", dev, err)
+		ctx.Warn(msg)
+		return err
+	}
+
+	m := parseNicAttrEthtool(&out)
+	nic.updateNicAttrEthtool(m)
+
+	return nil
```

Comment:
I think it would be better to just merge this into the `netDeviceCapabilities()` function. Then you can re-use the `out` and `path` variables from that function instead of calling again `exec.LookPath` and creating another `bytes.Buffer`